From c3b1622e1a210e21ddf2187e6efe40b34518e86a Mon Sep 17 00:00:00 2001
From: Ranjani Vaidyanathan <Ranjani.Vaidyanathan@nxp.com>
Date: Fri, 9 Feb 2018 13:04:23 -0600
Subject: [PATCH 3365/5242] MLK-17561-4 ASoC:vpu: Update to the latest SCFW
 API based on commit 97b8a6ee

commit  e77cbf5bc5f2c64572c7c4d95d040d2b5e6e49ae from
https://source.codeaurora.org/external/imx/linux-imx.git

commit 97b8a6eed4eee19ec8a60dedfffc2f5f3d8933c5
Author: Chuck Cannon <chuck.cannon@freescale.com>
Date:   Tue Feb 6 08:54:16 2018 -0600

Add unique ID API call. Required to get info needed for SECO fuse
programming. Added info command to DM.

Signed-off-by: Ranjani Vaidyanathan <Ranjani.Vaidyanathan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/vpu-malone/mxc_vpu-malone.c |   14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/drivers/mxc/vpu-malone/mxc_vpu-malone.c b/drivers/mxc/vpu-malone/mxc_vpu-malone.c
index d433254..8ce2787 100755
--- a/drivers/mxc/vpu-malone/mxc_vpu-malone.c
+++ b/drivers/mxc/vpu-malone/mxc_vpu-malone.c
@@ -143,12 +143,13 @@ static int setVPUPwr(sc_ipc_t ipcHndl,
 	}
 
 	/* Power on or off PID0, DEC, ENC */
-	sciErr = sc_pm_set_resource_power_mode(ipcHndl, SC_R_VPU_PID0, pm);
+	sciErr = sc_pm_set_resource_power_mode(ipcHndl, SC_R_VPU, pm);
 	if (sciErr != SC_ERR_NONE)
 	{
-		dev_err(vpu_dev, "--- sc_pm_set_resource_power_mode(SC_R_VPU_PID0,%d) SCI error! (%d)\n", pm, sciErr);
+		dev_err(vpu_dev, "--- sc_pm_set_resource_power_mode(SC_R_VPU,%d) SCI error! (%d)\n", pm, sciErr);
 		goto setVPUPwrExit;
 	}
+#if 0
 	/* FIXME: no need of PID1-7? */
 	sciErr = sc_pm_set_resource_power_mode(ipcHndl, SC_R_VPU_PID1, pm);
 	if (sciErr != SC_ERR_NONE)
@@ -192,16 +193,17 @@ static int setVPUPwr(sc_ipc_t ipcHndl,
 		dev_err(vpu_dev, "--- sc_pm_set_resource_power_mode(SC_R_VPU_PID7,%d) SCI error! (%d)\n", pm, sciErr);
 		goto setVPUPwrExit;
 	}
+#endif
 	sciErr = sc_pm_set_resource_power_mode(ipcHndl, SC_R_VPU_DEC_0, pm);
 	if (sciErr != SC_ERR_NONE)
 	{
-		dev_err(vpu_dev, "--- sc_pm_set_resource_power_mode(SC_R_VPU_DEC,%d) SCI error! (%d)\n", pm, sciErr);
+		dev_err(vpu_dev, "--- sc_pm_set_resource_power_mode(SC_R_VPU_DEC_0,%d) SCI error! (%d)\n", pm, sciErr);
 		goto setVPUPwrExit;
 	}
 	sciErr = sc_pm_set_resource_power_mode(ipcHndl, SC_R_VPU_ENC_0, pm);
 	if (sciErr != SC_ERR_NONE)
 	{
-		dev_err(vpu_dev, "--- sc_pm_set_resource_power_mode(SC_R_VPU_ENC,%d) SCI error! (%d)\n", pm, sciErr);
+		dev_err(vpu_dev, "--- sc_pm_set_resource_power_mode(SC_R_VPU_ENC_0,%d) SCI error! (%d)\n", pm, sciErr);
 		goto setVPUPwrExit;
 	}
 
@@ -227,10 +229,10 @@ static sc_pm_power_mode_t getVPUPwr(sc_ipc_t ipcHndl)
 		return (rv);
 	}
 
-	sciErr = sc_pm_get_resource_power_mode(ipcHndl, SC_R_VPU_PID0, &rv);
+	sciErr = sc_pm_get_resource_power_mode(ipcHndl, SC_R_VPU, &rv);
 	if (sciErr != SC_ERR_NONE)
 	{
-		dev_err(vpu_dev, "--- sc_pm_get_resource_power_mode(SC_R_VPU_PID0) SCI error! (%d)\n", sciErr);
+		dev_err(vpu_dev, "--- sc_pm_get_resource_power_mode(SC_R_VPU) SCI error! (%d)\n", sciErr);
 		rv = -1;
 	}
 
-- 
1.7.9.5

